### ISSUE TITLE
Help text wrongly states that the default `--builddir` is `dist`

### ISSUE THREAD

**Issue by andreasabel** [haskell/cabal#7924]
Help text for `cabal install` still claims that the default `--builddir` is `dist` (rather than `dist-newstyle`).
```
$ cabal install --help | grep -B1 dist
 --builddir=DIR                 The directory where Cabal puts generated build
                                files (default dist)
```

**Comment by phadej** [comment]
This is indeed embarrassing. If you grep the code, you will notice that the help message comes from `Cabal`, not `cabal-install`: https://github.com/haskell/cabal/blob/b6d051689518a380a9f5b29a34ad734a22546855/Cabal/src/Distribution/Simple/Setup.hs#L2159-L2171 (`defaultDistPref` is defined in the same file).

I *think* that the underlying reason is the same as why there is 

```
% cabal build --help|grep libdir 
 --libdir=DIR                   installation directory for libraries
 --libsubdir=DIR                subdirectory of libdir in which libs are
 --dynlibdir=DIR                installation directory for dynamic libraries
```

etc parameters in `cabal v2-build` which shouldn't be there, as they don't have any effect. However `Cabal`'s `builddir` is repurposed to do reasonable thing for `cabal-install`. It's somewhat hard to trace where from exactly `builddir` configuration param is 

TL;DR, this is IMO a symptom of v2 commands not having sane, independent cli-argument implementation. There are plenty of such small annoyances. I guess the symptoms could be fixed one-by-one: in this case parameterize `optionDistPref` by default location and use `dist-newstyle` in `cabal-install`); however fixing the underlying cause needs a very determined person.

**Comment by phadej** [comment]
This is not oneline change, multiple files, in separate packages have to be changed. (This shouldn't be that hard, but that is what I mention as a hard underlying problem).
